odoo/tools/mail.py

```diff
@@ -36,6 +37,27 @@
      'data-publish', 'data-id', 'data-res_id', 'data-member_id', 'data-view-id'
      ])
 
+# allow base64 image
+_image_dataurl = re.compile(r'(data:image/(.+));base64,([^"\']*)', re.I)
+_is_possibly_malicious_scheme = re.compile(r'^(?:javascript|jscript|livescript|vbscript|data|about|mocha):', re.I).search
+def _is_javascript_scheme(s):
+    if _image_dataurl.search(s):
+        return None
+    return _is_possibly_malicious_scheme(s)
+lxml.html.clean._is_javascript_scheme = _is_javascript_scheme
+
+
+def extractImageBase64(body, nameGenerator):
+    boundaries = dict()
+    def base64ToBoundary(match):
+        boundary = boundaries.get(match.group(0))
+        if not boundary:
+            fileType = match.group(2)
+            name = nameGenerator('boundary-%s.%s' % (len(boundaries), fileType), match.group(3))
+            boundary = (name, match.group(3))
```

Comment:
"boundary-3-jpeg" doesn't seem like a very useful attachment name, in the world of useless attachment names.